### PR TITLE
Improves: the performance and Architecture of My Site Dashboard

### LINKED ISSUE #18956: Dashboard Personalization: Handle No Cards in dashboard
Implement empty views when all cards have been hidden
Note: Take into consideration the possibility of Next Steps/Quick start card

Desing ref: YSMw2nbFAgPpjZLLbYvTfT-fi-0_1

### PR DESCRIPTION
**Pull request by AjeshRPai**
## Description
This PR improves the performance of MySiteDashboard by Removing the recyclerview inside the recyclerview used in My Site dashboard screen. Earlier we were using  `CardsAdapter` inside the `MySiteAdapter`, With this PR all the logic inside CardsAdapter has been shifted to MySiteAdapter. Also, the subclass  `DashboardCard`  has been removed. All the subclassess of`DashboardCard` class will directly inherit `Card` class

### 1. Smoke tests
- Go to mysite dashboard
- Verify that all the cards are shown as expected and the tracking is as before
- Tap the Personalise card on the dashboard 
- Hide each dashboard card and verify that it is not shown on dashboard
- Click on each card and verify that that click events are tracked as expected

### 2. Debug Overdraw performance check
- Enable [Debug Overdraw](https://developer.android.com/topic/performance/rendering/inspect-gpu-rendering#debug_overdraw)
- Go to MySiteDashboard 
- Verify that most of the items in Dashboard are drawn only once or twice - shown by Blue or Green visual element 

## Regression Notes
1. Potential unintended areas of impact
My Site dashboard is not shown as expected

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Unit tests and Manual testing

3. What automated tests I added (or what prevented me from doing so)
Unit test

PR submission checklist:
- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [x] High contrast.
- [x] Talkback.
- [x] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] Large and small screen sizes. (Tablet and smaller phones)
- [x] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Comment by RenanLukas** [comment]
Thanks for the PR, @AjeshRPai! I'm getting an error while trying to compile and it seems like it's the same problem reported by CI:
```
[2023-09-19T06:58:43Z] public final class NoCardsMessageViewHolder extends CardViewHolder<org.wordpress.android.databinding.NoCardsMessageBinding> {
[2023-09-19T06:58:43Z]                                                     ^
[2023-09-19T06:58:43Z]   symbol: class CardViewHolder
[2023-09-19T06:58:43Z] 
[2023-09-19T06:58:43Z] > Task :WordPress:kaptJetpackJalapenoDebugKotlin FAILED
```

Considering your changes, I suppose the fix would be to make `NoCardsMessageViewHolder` inherit from `MySiteCardAndItemViewHolder` instead of `CardViewHolder` since this one was removed.

After fixing the above locally, I came across the following compilation error but I'm not sure what would be the replacement for the removed `MySiteCardAndItem.Type.DASHBOARD_CARDS`:
image

**Comment by RenanLukas** [comment]
Thanks for taking care of this, @AjeshRPai. I've tested the overdraw and it's looking a lot better than before 👍 

### Before
Screen Shot 2023-09-20 at 2 55 41 PM

### After
Screen Shot 2023-09-20 at 3 02 12 PM

**Comment by AjeshRPai** [comment]
Hey @pachlava 

Pinging you to let you know that we have made some changes in the `ActivityLogCard` E2E tests with this commit https://github.com/wordpress-mobile/WordPress-Android/pull/19199/commits/25e981f6a3f4576f748d14d04bb88677e06c3a04

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/BloggingPromptsCardTrackHelper.kt [hunk 1]
```diff
@@ -42,12 +41,12 @@ class BloggingPromptsCardTrackHelper @Inject constructor(
     }
```
Comment (zwarm) [inline WordPress/src/main/java/org/wordpress/android/ui/mysite/BloggingPromptsCardTrackHelper.kt:42]: Nitpick: should probably eliminate one of these blank lines

**Review by zwarm — APPROVED** [review]
👏  Nice work @AjeshRPai . Looking so much better; thanks for taking this one. Can't wait to see what future improvements with bring. I left one teeny-tiny nit, but it's not stopping from approval.

I'm not going to merge because we may want to wait for release to be merged back into trunk first; then get trunk merged into here. wdyt?

**Pull request merged**